wdl-engine/src/stdlib/contains_key.rs

```diff
@@ -0,0 +1,354 @@
+            ]
+        },
+    )
+}
+
+#[cfg(test)]
+mod test {
+    use wdl_analysis::types::PrimitiveTypeKind;
+    use wdl_analysis::types::StructType;
+    use wdl_ast::version::V1;
+
+    use crate::v1::test::TestEnv;
+    use crate::v1::test::eval_v1_expr;
+
+    #[test]
+    fn contains_key() {
+        let mut env = TestEnv::default();
+
+        let bar_ty = env
+            .types_mut()
+            .add_struct(StructType::new("Bar", [("baz", PrimitiveTypeKind::String)]));
+
+        env.insert_struct("Bar", bar_ty);
+
+        let foo_ty = env
+            .types_mut()
+            .add_struct(StructType::new("Foo", [("bar", bar_ty)]));
+
+        env.insert_struct("Foo", foo_ty);
+
+        let value = eval_v1_expr(&mut env, V1::Two, "contains_key({}, 1)").unwrap();
+        assert!(!value.unwrap_boolean());
+
+        let value =
+            eval_v1_expr(&mut env, V1::Two, "contains_key({ 1: 2, None: 3}, None)").unwrap();
+        assert!(value.unwrap_boolean());
+
+        let value = eval_v1_expr(&mut env, V1::Two, "contains_key({ 1: 2 }, 1)").unwrap();
+        assert!(value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key({ 'foo': 1, 'bar': 2, 'baz': 3 }, 'qux')",
+        )
+        .unwrap();
+        assert!(!value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key({ 'foo': 1, 'bar': 2, 'baz': 3 }, 'baz')",
+        )
+        .unwrap();
+        assert!(value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key(object { foo: 1, bar: 2, baz: 3 }, 'qux')",
+        )
+        .unwrap();
+        assert!(!value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key(object { foo: 1, bar: 2, baz: 3 }, 'baz')",
+        )
+        .unwrap();
+        assert!(value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key({ 'foo': 1, 'bar': 2, 'baz': 3 }, ['qux'])",
+        )
+        .unwrap();
+        assert!(!value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key({ 'foo': 1, 'bar': 2, 'baz': 3 }, ['baz'])",
+        )
+        .unwrap();
+        assert!(value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key(object { foo: 1, bar: 2, baz: 3 }, ['qux'])",
+        )
+        .unwrap();
+        assert!(!value.unwrap_boolean());
+
+        let value = eval_v1_expr(
+            &mut env,
+            V1::Two,
+            "contains_key(object { foo: 1, bar: 2, baz: 3 }, ['baz'])",
+        )
+        .unwrap();
+        assert!(value.unwrap_boolean());
+
+        let value = eval_v1_expr(
```

Comment:
Should we also test `baz` after this?

Reply:
So that I better understand, what would be the expression to test?

Reply:
We do test `['bar', 'baz']` further below (should be true), if that's what you mean.

Otherwise, a test for `['baz']` would be the same as a test for `['qux']` here (neither are keys of the `Foo` struct); I can change the test to use `baz` instead, though.

Reply:
I'm going to merge this as-is and put up a follow-up PR if warranted after discussion.